modules/new_serverpod_auth/serverpod_auth_profile/serverpod_auth_profile_server/lib/src/business/user_profiles.dart

```diff
@@ -0,0 +1,155 @@
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_auth_profile_server/serverpod_auth_profile_server.dart';
+
+/// Business logic for handling user profiles
+abstract final class UserProfiles {
+  /// Creates a new user and stores it in the database.
+  static Future<UserProfile> createUserProfile(
+    final Session session,
+    UserProfile userProfile,
+  ) async {
+    userProfile = (await UserProfileConfig.current.onBeforeUserProfileCreated
+            ?.call(session, userProfile)) ??
+        userProfile;
+
+    userProfile = userProfile.id == null
+        ? await UserProfile.db.insertRow(session, userProfile)
+        : await UserProfile.db.updateRow(session, userProfile);
+
+    await UserProfileConfig.current.onAfterUserProfileCreated?.call(
+      session,
+      userProfile,
+    );
+
+    return userProfile;
+  }
+
+  /// Finds a user profile by its email address.
+  ///
+  /// Returns `null` if no user profile is found.
+  static Future<UserProfile?> maybeFindUserProfileByEmail(
+    final Session session,
+    final String email,
+  ) async {
+    return await UserProfile.db.findFirstRow(
```

Comment:
This needs to be changed to find all rows and then handle the case where it finds 2 or more (as the email is not unique).

Return value TBD, should it just be `null`, or a special error case?

Reply:
changed to only return profile if there is 1 match.